### DISCUSSION TITLE
Timezone at specified time

### DISCUSSION THREAD

**Discussion by Sytten** [Kijewski/tzdb#99] [Q&A] [answered]
Hi there 👋 
I am trying to move away from usage of `localtime_r` and I stubbled on this library from the `chrono` implementation and the recommendation from `tz-rs`.
I use the `time` crate `local_offset_at` a lot and it seems like it was reimplemented in chrono either with `find_local_time_type_from_local` for unix or a custom implementation for windows (https://github.com/chronotope/chrono/pull/677/files).
I am wondering if that could be something to be added in this library or not?
Thanks!

**Comment by Sytten** [comment]
Ok I think I was really confused, you just need a very simple wrapper function to use the local timezone function, transform it to an utc offset and apply it to your timezone. I don't think there is a need to add it to this lib, but an example using the `time` crate could be cool. I am trying to push the dev to move toward a rust implementation: https://github.com/time-rs/time/issues/493

**Comment by Kijewski** [comment, marked as answer]
Hi Émile! So, you want to project a timestamp into the local time zone? Then [`tz::UtcDateTime::project()`](https://docs.rs/tz-rs/0.6.12/tz/datetime/struct.UtcDateTime.html#method.project) and/or [`tz::DateTime::project()`](https://docs.rs/tz-rs/0.6.12/tz/datetime/struct.DateTime.html#method.project) should be what you are looking for:

```rust
let tz = tzdb::local_tz().unwrap();
let utc_ts = tz::UtcDateTime::new(2022, 8, 8, 14, 49, 33, 369777135).unwrap();
let local_ts = utc_ts.project(tz).unwrap();
dbg!(
    // "2022-08-08T14:49:33.369777135Z"
    utc_ts.to_string(),
    // "2022-08-08T16:49:33.369777135+02:00"
    local_ts.to_string(),
    // "CEST"
    local_ts.local_time_type().time_zone_designation(),
    // 7200
    local_ts.local_time_type().ut_offset(),
);
```

**Reply by Sytten** [reply]
Mainly wanted it to work with the time crate, I published my implementation on the linked issue. Thanks for the reply though!
